File: scripts/add_branch_user.py
import prisma
import bcrypt

# San Jose branch user credentials
username = "sanjose_admin"
password = "sanjoseadmin"
branch = "sanjose"
role = "branch"

hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')

async def main():
    await prisma.connect()
    await prisma.models.BranchUser.create(
        data={
            "username": username,
            "password": hashed_password,
            "branch": branch,
            "role": role
        }
    )
    await prisma.disconnect()

if __name__ == "__main__":
    import asyncio
    asyncio.run(main())

File: scripts/add_branch_user.js
const { PrismaClient } = require('@prisma/client');
const bcrypt = require('bcryptjs');

const prisma = new PrismaClient();

async function main() {
  const username = 'sanjose_admin';
  const password = 'sanjoseadmin';
  const branch = 'sanjose';
  const role = 'branch';

  const hashedPassword = await bcrypt.hash(password, 10);

  // Upsert to avoid duplicates
  await prisma.branchUser.upsert({
    where: { username },
    update: {
      password: hashedPassword,
      branch,
      role,
    },
    create: {
      username,
      password: hashedPassword,
      branch,
      role,
    },
  });

  console.log('San Jose branch user added/updated successfully.');
}

main()
  .catch(e => {
    console.error(e);
    process.exit(1);
  })
  .finally(() => prisma.$disconnect());
